sd3_infer.py

```diff
@@ -171,42 +230,65 @@ def __init__(self, model):
 VAEFile = None  # "models/sd3_vae.safetensors"
 # Optional init image file path
 INIT_IMAGE = None
+# ControlNet
+CONTROLNET_COND_IMAGE = None
 # If init_image is given, this is the percentage of denoising steps to run (1.0 = full denoise, 0.0 = no denoise at all)
 DENOISE = 0.6
 # Output file path
 OUTDIR = "outputs"
 # SAMPLER
-# SAMPLER = "euler"
 SAMPLER = "dpmpp_2m"
+# MODEL FOLDER
+MODEL_FOLDER = "models"
 
 
 class SD3Inferencer:
+
+    def __init__(self):
+        self.verbose = False
+
     def print(self, txt):
         if self.verbose:
             print(txt)
 
-    def load(self, model=MODEL, vae=VAEFile, shift=SHIFT, verbose=False):
+    def load(
+        self,
+        model=MODEL,
+        vae=VAEFile,
+        shift=SHIFT,
+        controlnet_ckpt=None,
+        model_folder: str = MODEL_FOLDER,
+        text_encoder_device: str = "cpu",
+        verbose=False,
+        load_tokenizers: bool = True,
+    ):
         self.verbose = verbose
         print("Loading tokenizers...")
         # NOTE: if you need a reference impl for a high performance CLIP tokenizer instead of just using the HF transformers one,
         # check https://github.com/Stability-AI/StableSwarmUI/blob/master/src/Utils/CliplikeTokenizer.cs
         # (T5 tokenizer is different though)
         self.tokenizer = SD3Tokenizer()
-        print("Loading OpenAI CLIP L...")
-        self.clip_l = ClipL()
-        print("Loading OpenCLIP bigG...")
-        self.clip_g = ClipG()
-        print("Loading Google T5-v1-XXL...")
-        self.t5xxl = T5XXL()
+        if load_tokenizers:
+            print("Loading Google T5-v1-XXL...")
+            self.t5xxl = T5XXL(model_folder, text_encoder_device, torch.float32)
+            print("Loading OpenAI CLIP L...")
+            self.clip_l = ClipL(model_folder)
+            print("Loading OpenCLIP bigG...")
+            self.clip_g = ClipG(model_folder, text_encoder_device)
         print(f"Loading SD3 model {os.path.basename(model)}...")
-        self.sd3 = SD3(model, shift, verbose)
+        self.sd3 = SD3(model, shift, controlnet_ckpt, verbose, "cuda")
```

Comment:
Same comment as earlier. Models are to be loaded into and offloaded from GPU just before and after they are used, right?